apiary-metastore-listener/src/test/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListenerTest.java

```diff
@@ -105,11 +115,138 @@ public void onInsert() throws MetaException {
   }
 
-  // TODO: tests for other onXxx() methods
+  @Test
+  public void onAddPartition() throws MetaException {
+    Table table = new Table();
+    table.setTableName(TABLE_NAME);
+    table.setDbName(DB_NAME);
+
+    AddPartitionEvent event = mock(AddPartitionEvent.class);
+    when(event.getStatus()).thenReturn(true);
+    when(event.getTable()).thenReturn(table);
+
+    List<Partition> partitions = new ArrayList<>();
+    partitions
+        .add(new Partition(Arrays.asList("col_1", "col_2", "col_3"), DB_NAME, TABLE_NAME, 0, 0, new StorageDescriptor(),
+            ImmutableMap.of()));
+
+    when(event.getPartitionIterator()).thenReturn(partitions.iterator());
+
+    snsListener.onAddPartition(event);
+    verify(snsClient).publish(requestCaptor.capture());
+    PublishRequest publishRequest = requestCaptor.getValue();
+
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"ADD_PARTITION\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"partition\":\"[col_1, col_2, col_3]\"}"));
+  }
+
+  @Test
+  public void onDropPartition() throws MetaException {
+    Table table = new Table();
+    table.setTableName(TABLE_NAME);
+    table.setDbName(DB_NAME);
+
+    DropPartitionEvent event = mock(DropPartitionEvent.class);
+    when(event.getStatus()).thenReturn(true);
+    when(event.getTable()).thenReturn(table);
+
+    List<Partition> partitions = new ArrayList<>();
+    partitions
+        .add(new Partition(Arrays.asList("col_1", "col_2", "col_3"), DB_NAME, TABLE_NAME, 0, 0, new StorageDescriptor(),
+            ImmutableMap.of()));
+
+    when(event.getPartitionIterator()).thenReturn(partitions.iterator());
+
+    snsListener.onDropPartition(event);
+    verify(snsClient).publish(requestCaptor.capture());
+    PublishRequest publishRequest = requestCaptor.getValue();
+
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"DROP_PARTITION\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"partition\":\"[col_1, col_2, col_3]\"}"));
+  }
+
+  @Test
+  public void onDropTable() throws MetaException {
+    Table table = new Table();
+    table.setTableName(TABLE_NAME);
+    table.setDbName(DB_NAME);
+
+    DropTableEvent event = mock(DropTableEvent.class);
+    when(event.getStatus()).thenReturn(true);
+    when(event.getTable()).thenReturn(table);
+
+    snsListener.onDropTable(event);
+    verify(snsClient).publish(requestCaptor.capture());
+    PublishRequest publishRequest = requestCaptor.getValue();
+
+    assertThat(publishRequest.getMessage(),
+        is("{\"protocolVersion\":\""
+            + PROTOCOL_VERSION
+            + "\",\"eventType\":\"DROP_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\"}"));
+  }
+
+  @Test
+  public void onAlterPartition() throws MetaException {
+
```

Comment:
Here's the blank line put in to test me right?

Reply:
😃 Removed !